app/controllers/subscriptions_controller.rb

```diff
@@ -25,23 +28,39 @@ def create
     render 'new'
   end
 
+  def renew
+    subscription = Subscription.find_or_initialize_by_token(params[:subscription_id])
+    @subscription = SubscriptionPresenter.new(subscription)
```

Comment:
@erbridge 
`Could this also be using fetch_subscription_from_token?`
This doesn’t seem to be the case.
We’ve had an attempt at using the `fetch_subscription_from_token` method. however the subscription can’t be found when by using `find_and_verify_by_token` (which is being implemented within `fetch_subscription_from_token`).
It seems like the renew action works for two use cases: Renewing a subscription prior to expiry (e.g user clicks link before expiry) and renewing after a subscription has expired (e.g user clicks link after expiry date).
I imagine that’s where the need for the `find_or_initialize_by_token` method came from.
We’d attempted the refactoring and the tests error on not being able to find a subscription (i.e `@subscription` is nil) which is why we feel it might a case of renew attending to two different use cases. Any suggestions or feedback on this though would be appreciated.

`Or could fetch_subscription_from_token add the presenter wrapper?`
Not sure how this would work.
The `unsubscribe` method doesn’t implement the presenter and `fetch_subscription_from_token` is a before action for this method. Would adding the presenter mean that the function of this method would have to change? Would also appreciate any suggestions or feedback on this though.

Reply:
:+1: